packages/local_auth/example/ios/RunnerTests/FLTLocalAuthPluginTests.m

```diff
@@ -186,4 +186,47 @@ - (void)testFailedAuthWithoutBiometrics {
   [self waitForExpectationsWithTimeout:kTimeout handler:nil];
 }
 
+- (void)testLocalizedFallbackTitle {
+  FLTLocalAuthPlugin* plugin = [[FLTLocalAuthPlugin alloc] init];
+  id mockAuthContext = OCMClassMock([LAContext class]);
+  plugin.authContextOverrides = @[ mockAuthContext ];
+
+  const LAPolicy policy = LAPolicyDeviceOwnerAuthentication;
+  NSString* reason = @"a reason";
+  NSString* localizedFallbackTitle = @"a title";
+  OCMStub([mockAuthContext canEvaluatePolicy:policy error:[OCMArg setTo:nil]]).andReturn(YES);
+
+  // evaluatePolicy:localizedReason:reply: calls back on an internal queue, which is not
+  // guaranteed to be on the main thread. Ensure that's handled correctly by calling back on
+  // a background thread.
+  void (^backgroundThreadReplyCaller)(NSInvocation*) = ^(NSInvocation* invocation) {
+    void (^reply)(BOOL, NSError*);
+    [invocation getArgument:&reply atIndex:4];
+    dispatch_async(dispatch_get_global_queue(QOS_CLASS_BACKGROUND, 0), ^{
+      reply(NO, [NSError errorWithDomain:@"error" code:99 userInfo:nil]);
+    });
+  };
+  OCMStub([mockAuthContext evaluatePolicy:policy localizedReason:reason reply:[OCMArg any]])
+      .andDo(backgroundThreadReplyCaller);
+
+  FlutterMethodCall* call =
+      [FlutterMethodCall methodCallWithMethodName:@"authenticate"
+                                        arguments:@{
+                                          @"biometricOnly" : @(NO),
+                                          @"localizedReason" : reason,
+                                          @"localizedFallbackTitle" : localizedFallbackTitle,
+                                        }];
+
+  XCTestExpectation* expectation = [self expectationWithDescription:@"Result is called"];
+  [plugin handleMethodCall:call
+                    result:^(id _Nullable result) {
+                      XCTAssertTrue([NSThread isMainThread]);
+                      XCTAssertTrue([result isKindOfClass:[NSNumber class]]);
+                      OCMVerify([mockAuthContext setLocalizedFallbackTitle:localizedFallbackTitle]);
+                      XCTAssertFalse([result boolValue]);
+                      [expectation fulfill];
+                    }];
+  [self waitForExpectationsWithTimeout:kTimeout handler:nil];
+}
+
```

Comment:
There should be a test that when no fallback is passed, `setLocalizedFallbackTitle:` is not called at all, since part of what this PR is fixing is the call with `@""` that disabled the fallback.

Reply:
@stuartmorgan thank you for the fast feedback.
The previous 4 tests are called with arguments:
```
arguments:@{
  @"biometricOnly" : @(NO), // or YES
  @"localizedReason" : reason,
}];
```
Should I add `OCMVerify([mockAuthContext setLocalizedFallbackTitle:nil]);` to each of them?

Reply:
I would add a specific new test whose purpose is to make the specific assertion; unit tests should generally be as discuss as possible.

(You're right that it should asset that it's called with nil rather than not called; I misremembered the final code.)